plugins/outputs/sql/README.md

```diff
@@ -110,7 +110,7 @@ through the convert settings.
   #  text                 = "TEXT"
   #  timestamp            = "TIMESTAMP"
   #  defaultvalue         = "TEXT"
-  #  unsigned             = "UNSIGNED"
+  #  unsigned             = "INT UNSIGNED"
```

Comment:
It seems like this is a more valid default. I understand the SQL standard doesn't have unsigned integers and it's an extension that some databases (notably mysql) added. Since this plugin tries to generate standard sql that any database can understand, it's probably even more appropriate to have `unsigned = "INT"` here.

We need to be careful changing the default at all now though. For example, if someone uses this plugin with a database that understands just `UNSIGNED` and we change it, when they upgrade it could break for them. Maybe we shouldn't change the default unless we can show it never worked for commonly used databases.

Reply:
### mysql

various integer numeric types via `type unsigned`
https://dev.mysql.com/doc/refman/8.0/en/integer-types.html

### clickhouse

various unsigned integers types via `UInt#`
https://clickhouse.com/docs/en/sql-reference/data-types/int-uint/


### sqlite

integers use signed values, except for `UNSIGNED BIG INT`
https://www.sqlite.org/datatype3.html

### postgresql

no unsigned types
https://www.postgresql.org/docs/current/datatype.html


### microsoft sql

no unsigned types
https://social.msdn.microsoft.com/Forums/exchange/en-US/b91f934a-01a1-4146-9501-0dc228156e04/data-type-for-unsigned-integer?forum=transactsql

### snowflake

no unsigned types
https://docs.snowflake.com/en/sql-reference/data-types-numeric.html

Reply:
Before this PR, all the convert setting besides unsigned mapped 1:1 to a type. Unsigned was different in that it was a suffix always added to the integer setting. This PR makes the unsigned setting map to the unsigned type without adding the integer setting. I think this makes more sense, but anyone who used the unsigned setting before will be broken when they upgrade.

I don't think we can take this PR yet because while it fixes a problem for some users, it will create a problem for others.

Reply:
@reimda what is the way forward then?

Reply:
Can you check my understanding given it looks like it is MySQL and ClickHouse that are the two users of unsigned values:

For users of Clickhouse, unsigned is broken, so they need a fix before they can use unsigned values.

For users of MySQL, the default was essential `int unsigned`. By changing this, the default is still "int unsigned". However, in the event that the user changed integer to some other value, like `BIGINT`, then they would be getting `int unsigned` when they might have been expecting `bigint unsigned`?

Is that the match your understanding?

Reply:
Mysql and clickhouse have an unsigned column type, but they're not the only DBs that use the unsigned conversion. The conversion is needed by any database that's receiving unsigned values from telegraf, so all of them.

Take postgres for example. With the default settings, the plugin makes sql like `create table foo (fieldname integer unsigned)`. That's not valid for postgres and it will give you a syntax error about unsigned. To avoid this, postgres users of the plugin have something like `convert.unsigned = ""` in their config. I expect we have quite a few users in this situation because it's the only way to make postgres (and other DBs) write unsigned. If we accept this PR as is and those users upgrade, the plugin will start to produce sql for them like `create table foo (fieldname )` with no type at all which is also a syntax error.

At this point though I think the way forward is add another setting that lets users opt in to having convert.unsigned not be a suffix to convert.integer. Maybe the new setting should be `unsigned_is_prefix = true`. Then clickhouse users would set it to false and `convert.unsigned = "unit64"`and it would work as expected for them. We will need to put this in the 'driver-specific information' section in the readme.

When we make a future compatibility breaking release (2.0, 3.0, etc), we should make unsigned not be a suffix by default and get rid of this unsigned_is_prefix setting. At the same time, we should switch the default for convert.unsigned to INTEGER since that's valid ANSI/ISO SQL.

Reply:
We should put some more thought into the setting. unsigned_is_prefix is long and ugly, and we usually try to avoid bool settings because they're less clear to use and don't allow for future changes without adding another setting.

Maybe it should be `conversion_style = "unsigned_suffix"` and clickhouse would use `conversion_style = "literal"` or something

Reply:
To capture the breaking change, add it to #9478 and also add a comment to the code